server/src/test/java/org/elasticsearch/persistent/PersistentTasksCustomMetaDataTests.java

```diff
@@ -228,7 +242,66 @@ public void testBuilder() {
             assertEquals(changed, builder.isChanged());
             persistentTasks = builder.build();
         }
+    }
+
+    public void testMinVersionSerialization() throws IOException {
+        PersistentTasksCustomMetaData.Builder tasks = PersistentTasksCustomMetaData.builder();
+
+        Version minVersion = getFirstVersion();
+        final Version streamVersion = randomVersionBetween(random(), minVersion, getPreviousVersion(Version.CURRENT));
+        tasks.addTask("test_compatible_version", TestPersistentTasksExecutor.NAME,
+            new TestParams(null, randomVersionBetween(random(), minVersion, streamVersion),
+                randomBoolean() ? Optional.empty() : Optional.of("test")),
+            randomAssignment());
+        tasks.addTask("test_incompatible_version", TestPersistentTasksExecutor.NAME,
+            new TestParams(null, randomVersionBetween(random(), getNextVersion(streamVersion), Version.CURRENT),
+                randomBoolean() ? Optional.empty() : Optional.of("test")),
+            randomAssignment());
+        final BytesStreamOutput out = new BytesStreamOutput();
+        out.setVersion(streamVersion);
+        Set<String> features = new HashSet<>();
```

Comment:
The `features` are never set on the stream.

Reply:
meh. Thanks.